### PR TITLE
Fixes UI view

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
Fixes UI feature.
Only activate MVC View options if UI is enabled.

## Related issues
Addresses [issue #].

## Testing
Manual: Change, the following to true, then hit a url with the web browser `/Patient`
https://github.com/microsoft/fhir-server/blob/b376cef84d143f3a52348184a84cf6634e7c3183/src/Microsoft.Health.Fhir.Shared.Web/appsettings.json#L20

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by PTaladay — DISMISSED** [review]

File: src/Microsoft.Health.Fhir.Shared.Api/Registration/FhirServerServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -60,8 +60,7 @@ public static class FhirServerServiceCollectionExtensions
                 .AddNewtonsoftJson(options =>
                 {
                     options.SerializerSettings.DateParseHandling = Newtonsoft.Json.DateParseHandling.DateTimeOffset;
-                })
-                .AddRazorRuntimeCompilation();
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Shared.Api/Registration/FhirServerServiceCollectionExtensions.cs:64]: We need this call in there somewhere: 
image

File: src/Microsoft.Health.Fhir.Shared.Api/Modules/FeatureFlags/HtmlUi/HtmlUiFeatureModule.cs [hunk 2]
```diff
@@ -36,6 +39,12 @@ public void Load(IServiceCollection services)
                     .Singleton()
                     .AsSelf()
                     .AsService<TextOutputFormatter>();
+
+                // Adds provider to serve embedded razor views
+                services.Configure<MvcRazorRuntimeCompilationOptions>(options =>
+                {
+                    options.FileProviders.Add(new EmbeddedFileProvider(typeof(CodePreviewModel).Assembly));
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Shared.Api/Modules/FeatureFlags/HtmlUi/HtmlUiFeatureModule.cs:46]: I think we can add the razer view compilation here

```suggestion
                services.Configure<MvcRazorRuntimeCompilationOptions>(options =>
                {
                    options.FileProviders.Add(new EmbeddedFileProvider(typeof(CodePreviewModel).Assembly)).AddMvc().AddRazorRuntimeCompilation();
```

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HtmlOutputFormatter.cs [hunk 3]
```diff
@@ -67,58 +68,58 @@ public override async Task WriteResponseBodyAsync(OutputFormatterWriteContext co
...
-            var tempDataProvider = context.HttpContext.RequestServices.GetService<ITempDataProvider>();
+            IRazorViewEngine engine = context.HttpContext.RequestServices.GetService<IRazorViewEngine>();
+            ITempDataProvider tempDataProvider = context.HttpContext.RequestServices.GetService<ITempDataProvider>();
+
+            context.HttpContext.AllowSynchronousIO();
 
             var actionContext = new ActionContext(context.HttpContext, context.HttpContext.GetRouteData(), new ActionDescriptor());
 
-            using (TextWriter textWriter = context.WriterFactory(context.HttpContext.Response.Body, selectedEncoding))
-            {
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HtmlOutputFormatter.cs:76]: main changes to this file was un-nesting usings, and adding `AllowSynchronousIO();`

**Review by fhibf — APPROVED** [review]

**Review by mikaelweave — APPROVED** [review]
Tested locally - had no idea this existed. Cool. 😄

**Pull request merged**